_posts/2017-08-24-api-platform-en.md

```diff
@@ -0,0 +1,121 @@
+---
+layout: post
+title: Build an API with API Platform
+authors:
+    - rpierlot
+permalink: /en/build-an-api-with-api-platform/
+excerpt: "Api Platform defines itself as a « PHP framework to build modern web APIs ». Indeed, this tool will help us to rapidly build a rich and easily usable API."
+categories:
+    - PHP
+tags:
+    - php
+cover: /assets/2017-07-25-api-platform/cover.jpg
+---
+
+Api Platform defines itself as a « PHP framework to build modern web APIs ». Indeed, this tool will help us rapidly build a rich and easily usable API.
+Why reinvent the wheel? Api Platform comes with a bunch of _features_ like an automatic documentation, filters, sorts, and many more.  
+
+In this article, we're going to build an API using Api Platform, and talk about some of the features that comes with it. I will assume that API platform has already been installed in your project.
+
+## Let's build our API
+
+We'll create an API around a single resource: _movie._ Indeed, we will build something that adds, deletes and updates movies.
+
+The first thing we need to do is creating our data model. A movie is made of a title, a release date, actors, a director... Many properties known from everyone.
+If we go through the documentation, we can see that we can generate our entities with [Schema.org](http://schema.org). This allows us to use a common language to define our usual resources (_Book, Organization_, _Person_…), but it's also understood by search engine like Google or Yahoo.
+
+Good for us, there is an _Movie_ entity from _Schema.org_, with many interesting properties. For simplicity, we'll only work with a some of them.
+
+```yaml
+# app/config/schemas.yml
+types:
+  Movie:
+    parent: false
+    properties:
+      name: ~
+      dateCreated: ~
+      countryOfOrigin: { range: "Text" }
+```
+
+To generate our entity, we only need to execute the following command:
+
+```
+vendor/bin/schema generate-types src/ app/config/schema.yml
+```
+
+Yay ! If you go to src/AppBundle/Entity/Movie.php, we get an automatic generated entity that has all kinds of validation according to the types set in _Schema.org_.
```

Comment:
`we get an automatic generated entity` -> `we get an entity generated automatically`